src/hitchhiker/tree/core.clj

```diff
@@ -141,72 +215,80 @@
           a (object-array l)
           _ (dotimes [i l]
               (aset a i (last-key (nth children i))))
-          x (Arrays/binarySearch a 0 l key compare)]
+          x #?(:clj (Arrays/binarySearch a 0 l key compare)
+               :cljs (goog.array/binarySearch a key compare))]
```

Comment:
Unless we're 100% sure this is the exact same algorithm (in the way it interprets inclusive/exclusive bounds), I'd like to make sure this has a new generative test to cover its behavior.